packages/components/src/List/ListItem.tsx

```diff
@@ -9,7 +9,15 @@ import { Typography } from "../Typography";
 import { Markdown } from "../Markdown";
 import { Emphasis } from "../Emphasis";
 
-export interface ListItemProps {
+export interface BaseListItemProps {
+  /**
+   * The ID of the list item. This will be helpful to know the selected list
+   * items when a batch action is implemented.
```

Comment:
I know this is a copy/paste of an existing JSDoc, but I'm not sure I get what this is trying to say. don't we need an id all the time for arrays in React and not just for batch actions? actually what even is a batch action in this context...

Reply:
Yeah, good question 🤔 I didn't pay it much attention when I copied it over. I will update it to something that makes more sense.

Reply:
following up on this, even after all the time we've spent in List I have no idea what a "batch action" means in this context

my best guess is that you want to be using this in a selection context, and onClick you want to know the id of the thing you clicked to add it to a batch of items you wanna do _something_ with

however I'm of the opinion `List` is a really bad fit for selection style interactions, at least in its current state since we don't pass `id` as an arg to `props.onClick` meaning you'd have to dig through the `event.currentTarget` to get it which is gross.

Reply:
This all makes sense. Thank you for following up, with the other more major refactoring comments I missed addressing this one but will do so now